src/js/classes/DateTimeValidator.ts

```diff
@@ -0,0 +1,251 @@
+import PropTypes from 'prop-types';
+import {ensureValueIsDateObject} from '../helpers/developerInputValidators';
+import {getFieldValueFromKeyString, isInvalidDate} from '../helpers/generalHelpers';
+import BaseValidator from './BaseValidator';
+import {Dependent, ErrorFunction, InputType, MutationFunction} from '../types';
+
+const dateRegex = /^\d{4}-((0[1-9])|(1[0-2]))-(([0-2][0-9])|(3[0-1]))T(([0-1][0-9])|(2[0-3]))(:[0-5][0-9]){1,2}(.\d{3})?[Z]?$/s;
+
+/**
+ * A validator that allows you to validate datetime-local fields.
+ * @extends BaseValidator
+ */
+class DateTimeValidator extends BaseValidator<Date | string> {
+    public defaultInputType: InputType = 'datetime-local';
+    protected propType: PropTypes.Requireable<any> = PropTypes.object;
+
+    /**
+     * Validate a datetime input
+     * @param {Date|String} [defaultValue]
+     * @param {String} [defaultMsg]
+     * @param {MutationFunction} [mutationFunc]
+     * @param {ErrorFunction} [onError]
+     * @param {Dependent} [dependent]
+     */
+    constructor(defaultValue: Date | string = '',
+                defaultMsg: string = 'This field must contain a date/time',
+                mutationFunc?: MutationFunction,
+                onError?: ErrorFunction,
+                dependent?: Dependent) {
+        super(defaultValue, defaultMsg, mutationFunc, onError, dependent);
+
+        this.validateFuncs.push((value) => {
+            if (typeof value !== 'string') {
+                return {success: true, errorMsg: defaultMsg};
```

Comment:
Should we always return true when the value is not a string?  
I think a check to see if it is a Date object would make sense here

Reply:
value can only be a string or a Date here. if it's not a string, it must be a Date, at least if you work with TS and don't cast 'illegally'. If this library was only used with TS enabled, I would not worry about this. If a user uses Formifly with plain js tho... not sure. If you argue we should make this line more robust for these use cases, we should probably do that in a lot more places, too. I'm not sure if that's reasonable.